### ISSUE TITLE
WCAG 1.4.11 AA: Form input borders have insufficient contrast (Issue 20)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#372]
### Summary

Via Vision Australia assessment: August 2024

Impact: medium

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au.

### Steps to reproduce

[Contact digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/contact) (https://www.digital.gov.au/contact)

### Observed outcome

Within the contact form, the contrast between the light blue (#e6e9eb) input borders and the adjacent white (#ffffff) background does not meet the minimum ratio required. 

•	Current contrast ratio: 1.21:1
•	Required contrast ratio: 3:1

![Picture29](https://github.com/user-attachments/assets/6353352e-545d-40d3-97e2-83b075e76938)
 
### Code used
**HTML**
`<input type="text" value="" class="ct-textfield ct-input__element ct-theme-light form-text" #component_theme="light" data-drupal-selector="edit-name" id="edit-name" name="name" size="60" maxlength="255">`
**CSS**
```
.ct-input.ct-theme-light .ct-input__element {
border-color: #e6e9eb;
background-color: #ffffff;
}
```
### Why this matters
Users with low vision or colour perception disabilities may find it difficult or impossible to perceive user interface components that have a low colour contrast ratio with the background. 

### Expected outcome

Adjust the colours to ensure a minimum contrast ratio of 3:1 between the input borders and the adjacent background.

**Note 1**

Inactive (disabled) form controls are exempt from this success criterion. 

**Note 2**

The Colour Contrast Analyser (CCA) is a free tool that can be used to check the contrast of text and user interface components: [Colour Contrast Analyser (CCA)](file:///C:/Users/LiThomas/Downloads/Colour%20Contrast%20Analyser%20(CCA)) (https://www.tpgi.com/color-contrast-checker/)	

### Related issues

For other issues related to this component, please refer to:

•	Issue 6 – Label not programmatically associated with input
•	Issue 7 – Error messages not programmatically associated with input
•	Issue 12 – Inputs missing autocomplete attribute
•	Issue 32 – Required fields not indicated
•	Issue 37 – Success message not announced by screen readers

**Comment by CJE001** [comment]
**1.4.11 Non-text Contrast WCAG 2.1 Level AA**

The visual [presentation](https://www.w3.org/TR/WCAG21/#dfn-presentation) of the following have a [contrast ratio](https://www.w3.org/TR/WCAG21/#dfn-contrast-ratio) of at least 3:1 against adjacent colours:

•	User Interface Components: Visual information used to indicate states and boundaries of user interface components, except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;
•	Graphical Objects: Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.

Users with low vision can have difficulty identifying user interface components and determine which state they are in. This user group can also have difficulty interpreting graphics. Therefore, it is important to provide sufficient colour contrast for visual cues such as boundaries, states and graphical shapes.

**Comment by fionamorrison23** [comment]
@alanrako please read this issue. I'll create a ticket to update the design, separate to this ticket.

**Comment by fionamorrison23** [comment]
Closing this issue as it is related to the sub-theme. Form input borders in CivicTheme core have sufficient contrast to pass AA.
